src/Hedgehog/Range.fs

```diff
@@ -81,6 +81,13 @@ module Range =
 
         constantFrom zero lo hi
 
+    //
+    // Factories
+    //
+
+    let internal ofArray (xs : 'a array) =
```

Comment:
```suggestion
    let internal ofArray (xs : 'a array) : Range<int> =
```

* And since it's monomorphic, can we move it inside Gen.fs?

(You've already done a massive work, so I'm happy to do that change myself if you're tired.)

Reply:
I think this would make more sense as a `Range` function, but it can live wherever.

Reply:
Well, actually, I'm not sure it even makes sense as a function at all since it's used in [less than 3 places](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) but also [explicit is better than implicit](https://www.python.org/dev/peps/pep-0020/). But having it inside Gen.fs will make it easier for those [reading](https://www.goodreads.com/quotes/835238-indeed-the-ratio-of-time-spent-reading-versus-writing-is) the code, at least.

Reply:
`/grump on`

In the wikipedia article you linked, there's a "See Also" for the DRY principle, which I'm more likely to follow in my PRs. I've been in software for the better part of two decades and had never heard of the "Rule of three", as I find any repetition in my code abhorrent.

I don't see what a python manifesto has to do with an F# code base either :smile:

`/grump off`

Kidding aside though, I believe the best place for this function is in the `Range` module, since it's a constructor function for a `Range` that might be useful to users as well. The discussion around exposing this is separate though, so my vote would be to just leave it where it is but alas I am only a single vote :sweat_smile: 

As far as readability goes, plenty of other functions in `Gen.fs` call over to functions in `Range.fs` and readability hasn't been hindered for me in that regard. As long as the function being called has a sensible name, I can make a quick assumption as to its behavior as I'm reading. In my mind, putting functions where they don't belong was a big motivating factor for #247, where a lot of things had coalesced in the `Property.fs` file.

Perhaps we need @TysonMN to break the tie! :tada: 

@moodmosaic I appreciate the feedback as always :+1:

Reply:
>but alas I am only a single vote 

A single vote with enough power at this point. You've done a massive work. Feel free to leave it there as-is. I shouldn't be nitpicking that much. 😥